lib/kitchen/driver/openstack.rb

```diff
@@ -372,29 +368,6 @@ def hints_path
         Ohai::Config[:hints_path][0]
       end
 
-      def setup_ssh(server, state)
-        if config[:key_name]
-          info "Using OpenStack keypair <#{config[:key_name]}>"
-        end
-        info "Using public SSH key <#{config[:public_key_path]}>"
-        info "Using private SSH key <#{config[:private_key_path]}>"
-        state[:ssh_key] = config[:private_key_path]
-        do_ssh_setup(state, config, server) unless config[:key_name]
-      end
-
-      def do_ssh_setup(state, config, server)
```

Comment:
In your PR comments, you said that you're removing these methods because the transport does this for you, but I don't believe the transport actually pushes an SSH key out to an endpoint if it doesn't exist.